### PR TITLE
fix: edit messages for agent create and agent generte spec

### PR DESCRIPTION
**Pull request by jshackell-sfdc**
### What does this PR do?

Edits and adds a bit more info to the --help of "agent generate spec" and "agent create"

### What issues does this PR fix or reference?
@W-17417430@

### CODE REVIEW

File: messages/agent.create.md [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-The name of the agent.
+Name of the new agent.
 
 # examples
 
-- Create an Agent:
+- Create an agent called "Customer Support Agent" in an org with alias "my-org" using the specified agent spec file:
 
-  <%= config.bin %> <%= command.id %> --spec ./agent-spec.json
+  <%= config.bin %> <%= command.id %> --name "Customer Support Agent" --job-spec ./config/agentSpec.json --target-org my-org
```
Comment (WillieRuemmele) [inline messages/agent.create.md:25]: this command will fail, the name _is_ the API name, and can't contain spaces

Comment (jshackell-sfdc) [reply messages/agent.create.md:25]: Oh, huh.  Okay.  Seems weird to pass an API name.  So there's no way to get a more user-friendly name?

Comment (jshackell-sfdc) [reply messages/agent.create.md:25]: What I mean is that when they create the agent, in the UI the non-API name can never have spaces?

Comment (jshackell-sfdc) [reply messages/agent.create.md]: I updated the help.

File: messages/agent.create.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-# flags.job-spec.description
+# flags.job-spec.summary
 
-The agent spec file defines job titles and descriptions for the agent and can be created using the `sf agent create spec` command.
+Path to an agent spec file.
 
 # flags.name.summary
 
-The name of the agent.
+API name of the new agent.
```
Comment (shetzel) [inline messages/agent.create.md:19]: I believe this is actually the label, not the API name (aka developer name).

File: messages/agent.create.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
 # summary
 
-Create an Agent from an agent spec.
+Create an agent in your org from a local agent spec file.
```
Comment (shetzel) [inline messages/agent.create.md:3]: This is going to change with the v2 agent creator APIs that will allow "agent preview" behavior, meaning the metadata will not be created in the org but sent back in the response to be previewed by the customer.

File: messages/agent.create.md [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
 
 # description
 
-Create an Agent from an agent spec. Agent metadata is created in the target org and retrieved to the local project.
+To generate an agent spec file, run the "agent generate spec" CLI command, which outputs a JSON file with the list of jobs and descriptions that the new agent can perform. Then specify this generated spec file to the --job-spec flag of this command, along with the name of the new agent.
 
-# flags.job-spec.summary
+When this command finishes, your org contains the new agent, which you can then edit in the Agent Builder UI. The new agent already has a list of topics and actions that were automatically created from the list of jobs in the provided agent spec file. This command also retrieves all the metadata files associated with the new agent to your local DX project.
```
Comment (shetzel) [inline messages/agent.create.md:9]: This is true with the current implementation but not necessarily with the v2 implementation.

File: messages/agent.generate.spec.md [hunk 5]
```diff
@@ -1,41 +1,49 @@
 # summary
 
-Create an Agent spec.
+Generate an agent spec, which is the list of jobs that the agent performs.
 
 # description
 
-Create an Agent spec, which is a list of job titles and descriptions that the agent performs.
+When using Salesforce CLI to create an agent in your org, the first step is to generate the local JSON-formatted agent spec file with this command.
```
Comment (shetzel) [inline messages/agent.generate.spec.md:7]: This is true for the existing commands but not necessarily true with v2.  The v2 APIs allow agent creation from the fields used to generate a spec and not only from a spec.

**Review by shetzel — APPROVED** [review]
Note that the commands are not final and the way they work is going to change based on using the v2 APIs.

File: messages/agent.create.md (hunk 3 above)
Comment (jshackell-sfdc) [reply messages/agent.create.md:3]: Gotcha.  I was writing this with the Dec milestone in mind, so I will change it when you implement v2.  Okay?

File: messages/agent.create.md (hunk 4 above)
Comment (jshackell-sfdc) [reply messages/agent.create.md:9]: Ditto

File: messages/agent.generate.spec.md (hunk 5 above)
Comment (jshackell-sfdc) [reply messages/agent.generate.spec.md:7]: Ditto

File: messages/agent.create.md (hunk 2 above)
Comment (jshackell-sfdc) [reply messages/agent.create.md:19]: @WillieRuemmele said it was the API name.  I originally thought it was the label.

**Comment by jshackell-sfdc** [comment]
I'll update the --help after you implement v2.  I was writing this for the Dec milestone, in case Vivek wants to run --help on the commands in his demo.  :)

**Pull request merged**